src/crystal/system/unix/file.cr

```diff
@@ -105,6 +105,25 @@ module Crystal::System::File
     ret
   end
 
+  def self.readlink(path) : String
+    buf = Bytes.new 256
+    3.times do
+      ret = LibC.readlink(path, buf, buf.bytesize)
+      if ret == -1
+        raise Errno.new("readlink")
+      elsif ret == buf.bytesize
+        buf = Bytes.new(buf.bytesize * 4)
+      else
+        buf[ret] = 0_u8
+        return String.new(buf[0, ret])
+        break
```

Comment:
1. you can pass the actual bytesize as `String.new(buf, ret)`;
2. `break` is unreachable after `return`;

Reply:
BTW: `ret` could have a more explicit name, like `bytesize` for example.

Reply:
1. Should I use `String.new(buf.pointer, ret)`?  Add a new overload to String?  Keep String.new(buf[0, ret])? Do something else?

```
in src/crystal/system/unix/file.cr:120: no overload matches 'String.new' with types Slice(UInt8), Int64
Overloads are:
 - String.new(bytes : Bytes, encoding : String, invalid : Symbol | ::Nil = nil)
 - String.new(chars : ::Pointer(UInt8), bytesize, size = 0)
 - String.new(chars : ::Pointer(UInt8))
 - String.new(slice : Bytes)
 - String.new(capacity : Int, &block)
```

2. Changed to bytesize.  Not pushed yet.